### PR TITLE
Feature/update avro unmarshal

### PR DESCRIPTION
**Pull request by nshumoogum**
### What

Add functionality to the unmarshal method to handle objects nested within an object

### How to review

Make sure all unit tests pass and that the code makes logical sense.

Note: The commented out lines in the unit tests are placeholders for missing tests, these will be written when we have time to do so.

### Who can review

Team B

### CODE REVIEW

File: avro/test_data.go [hunk 1]
```diff
@@ -53,6 +53,61 @@ var testNestedArraySchema = `{
...
+		{
+		  "name": "footballer",
+			"type": {
+			  "name": "footballer-name",
+			  "type": "record",
+				"fields": [
+					{
+			  		"name": "surname",
+						"type": "string",
+            "default": ""
```
Comment (mattrout92) [inline avro/test_data.go]: spacing

File: avro/test_data.go [hunk 2]
```diff
@@ -53,6 +53,61 @@ var testNestedArraySchema = `{
...
+			 		},
+					{
+					  "name": "middle-name",
+						"type": "string",
+						"default": ""
+				  },
+					{
+						"name": "forename",
+						"type": "string",
+            "default": ""
```
Comment (mattrout92) [inline avro/test_data.go]: spacing

File: avro/test_data.go (hunk 1 above)
Comment (nshumoogum) [reply avro/test_data.go]: I saw this and I swear I used go fmt........ :(

Comment (nshumoogum) [reply avro/test_data.go]: Just realised its in a string so it won't format it, I have to do it manually :(

**Pull request merged**